src/protocol/sort/reshare.rs

```diff
@@ -81,52 +81,77 @@ impl<F: Field> Reshare<F> {
 
 #[cfg(test)]
 mod tests {
-    use futures::future::try_join_all;
+
     use proptest::prelude::Rng;
-    use rand::rngs::mock::StepRng;
 
+    use rand::thread_rng;
+
+    use crate::ff::Fp32BitPrime;
+    use crate::protocol::context::Context;
     use crate::{
-        ff::Fp31,
         helpers::Role,
         protocol::{sort::reshare::Reshare, QueryId, RecordId},
-        test_fixture::{make_contexts, make_world, share, validate_and_reconstruct, TestWorld},
+        test_fixture::{make_world, validate_and_reconstruct},
     };
 
+    use crate::test_fixture::Runner;
+
+    /// Validates that reshare protocol actually generates new shares using PRSS.
+    #[tokio::test]
+    async fn generates_unique_shares() {
+        let world = make_world(QueryId);
+
+        for &target in Role::all() {
+            let secret = thread_rng().gen::<Fp32BitPrime>();
+            let shares = world
+                .semi_honest(secret, |ctx, share| async move {
+                    let record_id = RecordId::from(0);
+
+                    // run reshare protocol for all helpers except the one that does not know the input
+                    if ctx.role() == target {
+                        // test follows the reshare protocol
+                        ctx.prss().generate_fields(record_id).into()
```

Comment:
Won't the effect be the same if you run reshare for this target?  Or, are you trying to test that it doesn't need to participate?

Reply:
I tried to ensure a naive `Reshare` implementation (that is - take the input and return it as output) won't pass this test. I think that was the reason for `for` loop inside the test before.